File: Mher/CS50/Week_1/credit.c
#include<stdio.h>
#include<cs50.h>
#include <math.h>

int main(void)
{
    long int n = get_long("Cart: ");
    long int save = n;
    int a = 0;
    
    while (n > 0)
    {
        a += n % 10;
        n = n / 10;
    
        if (n % 10 * 2 > 9)
        {
            a = a + ((n % 10 * 2) % 10) + 1;
            n = n / 10;
        }
        else
        {
            a += n % 10 * 2;
            n = n / 10;
        }
    }

    if (a % 10 == 0)
    {
        if (save / 10000000000000 == 34 || save / 10000000000000 == 37)
        {
            printf("AMEX\n");
        }
        else if (save / 1000000000000000 == 4 || save / 1000000000000000 == 4)
        {
            printf("VISA\n");
        }
        else if (save / 100000000000000 > 50 && save / 100000000000000 < 56)
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }         
    }    
    else
    {
        printf("INVALID\n");
    }
}

File: HaykAmirjanyan/pset6/credit/credit.py
from cs50 import get_int

card_number = get_int("Number your card: ")
number = [int(x) for x in str(card_number)]
number.reverse()
sum = 0
for i in range(0, len(number), 2):
    sum = sum + number[i]
for j in range(1, len(number), 2):
    sum += (number[j] * 2) //10 + (number[j] * 2) % 10
number.reverse()
if (number[0] == 4 and sum % 10 == 0 and (len(number) == 13 or len(number) == 16)):
    print("VISA")
elif (len(number) == 16 and number[0] == 5 and number[1] < 6 and sum % 10 == 0):
    print("MASTERCARD")
elif (len(number) == 15 and number[0] == 3 and (number[1] == 4 or number[1] == 7) and sum % 10 == 0):
    print("AMEX")
else:
    print("INVALID")